### PR TITLE
Timeout and color config, fix dimensions, pull in loaders.css, SSR, currentColor, spinnerName

### PR DESCRIPTION
**Pull request by merrywhether**
This PR addresses a few issues:

- #33 : Since the fade in is set as a CSS animation, I couldn't do full configurability, but added 2 smaller intervals of fadeIn timing (half second and quarter second). This includes transitioning away from the `noFadeIn` prop with a deprecation warning.
- #37 : This allows colorizing the spinners via props and inline styling for easier use. There were 2 spinners that weren't compatible with this approach because they relied on before/after pseudo-elements, and there are appropriate warnings for people who attempt to do that.
- #48 : A few of the spinners had misconfigured dimensions on the containing div, resulting in difficulty centering them via flexbox, etc. Their dimensions have been adjusted accordingly.
- #14 : Pulls in loaders.css for more spinners, with efforts to minimize package size growth. I omitted spinners from that repo that were very similar to pre-existing "native" spinners.
- #20 / #51 : Made style imports dynamic based on environment variable, allowing server side rendering without lots of hacks. This is similar to react's dynamic behavior based on `process.env.PRODUCTION` and is a specific enough variable that it shouldn't collide.

Additionally, updated tests, README, and demo site for all new features, and made changes to the test file to appease eslint (change to `.jsx`, convert tabs to spaces, etc) as well as organizational improvements (describe blocks to group related tests).

All changes were tested inside of a working app for functionality and proper centering alignment.

### CODE REVIEW

File: css/wandering-cubes.css [hunk 1]
```diff
@@ -22,9 +22,9 @@
 }
 
 @-webkit-keyframes sk-cubemove {
-  25% { -webkit-transform: translateX(22px) rotate(-90deg) scale(0.5) }
-  50% { -webkit-transform: translateX(22px) translateY(22px) rotate(-180deg) }
-  75% { -webkit-transform: translateX(0px) translateY(22px) rotate(-270deg) scale(0.5) }
+  25% { -webkit-transform: translateX(42px) rotate(-90deg) scale(0.5) }
+  50% { -webkit-transform: translateX(42px) translateY(42px) rotate(-180deg) }
+  75% { -webkit-transform: translateX(0px) translateY(42px) rotate(-270deg) scale(0.5) }
```
Comment (merrywhether) [inline css/wandering-cubes.css]: Not sure why these weren't on the same 42px scale and the non-vendor-prefixed keyframes, so I adjusted them. If there was a reason, I'm happy to revert.

File: css/three-bounce.css [hunk 2]
```diff
@@ -1,3 +1,7 @@
+.sk-three-bounce {
+  height: 18px;
+}
```
Comment (merrywhether) [inline css/three-bounce.css]: user-agent default styles added padding to the bottom of this div that affected proper vertical alignment.

**Comment by merrywhether** [comment]
Second commit refactors everything to be classes with anonymous children divs so that code can be DRYed up. Relies heavily on the suggestion in #27.

File: css/loaders-css.css [hunk 3]
```diff
@@ -0,0 +1,52 @@
+.ball-triangle-path > div,
```
Comment (merrywhether) [inline css/loaders-css.css]: Minor overrides to loaders.css styles to allow coloring under new scheme.

**Comment by merrywhether** [comment]
Okay, fully polished, and I'll stop messing with it.

**Comment by merrywhether** [comment]
This also addresses #53.

**Comment by merrywhether** [comment]
Since this represents a lot of changes, pulled in #19 as well, and maybe this should result in a 3.0 release.

**Comment by KyleAMathews** [comment]
This is awesome! Thanks for giving things a nice cleanup / upgrade! Adding colors support + loaders.css is 💯 

Merging and releasing a 3.0.

The next frontier I see for this module is adding tree shaking support. That or perhaps switch to a lerna monorepo approach where there's a package per spinner to keep the size of the package to a minimum.

**Pull request merged**